src/interpreter/mod.rs

```diff
@@ -593,7 +593,10 @@ where
                 Terminal::PkK(ref pk) => {
                     debug_assert_eq!(node_state.n_evaluated, 0);
                     debug_assert_eq!(node_state.n_satisfied, 0);
-                    let res = self.stack.evaluate_pk(&mut self.verify_sig, *pk);
+                    let res = match pk.single_key() {
+                        Some(pk) => self.stack.evaluate_pk(&mut self.verify_sig, *pk),
+                        None => None,
```

Comment:
Doing this is wrong. We should never encounter this because know the variant here is `single_key`. We should either use expect on `single_key` with a message stating "Musig keys cannot be parsed from Script"

Reply:
Ok. fixed as suggested.